inconspiquous/backend/qir.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+from io import StringIO
+from typing import Any
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import Operation
+
+try:
+    from pyqir import BasicQisBuilder, SimpleModule
+
+    pyqir_available = True
+except ImportError:
+    pyqir_available = False
+
+from inconspiquous.dialects import qir
+
+
+def print_qir(module: ModuleOp, output: StringIO) -> None:
+    """Print QIR assembly to output stream"""
+    if not pyqir_available:
+        output.write("; PyQIR not available. Install with: pip install pyqir\n")
+        return
+
+    backend = QIRBackend()
+    try:
+        qir_assembly = backend.emit_module(module)
+        output.write(qir_assembly)
+    except Exception as e:
+        output.write(f"; Error generating QIR: {e}\n")
+        output.write("; Module structure:\n")
+        for op in module.walk():
+            if hasattr(op, "name"):
+                output.write(f";   {op.name}\n")
```

Comment:
let's move this function into the `register_all_target`, here's how we do this in xDSL:

https://github.com/xdslproject/xdsl/blob/main/xdsl/xdsl_opt_main.py#L245

Note that the file itself assumes that the user importing it can import the dependencies, and we check for import errors at the point of use, raising the appropriate error depending on whether it's done in the command line or in another context.